### ISSUE TITLE
Udocker run for java docker image fails with "Could not find Java SE Runtime Environment" while docker run succeeds

### ISSUE THREAD

**Issue by tarunkumard** [indigo-dc/udocker#255]
Hi,

udocker run fails: for `udocker run denvazh/gatling` . but same run with `docker run denvazh/gatling`..here is error at udocker console

```
GATLING_HOME is set to /opt/gatling
Error: could not find libjava.so
Error: Could not find Java SE Runtime Environment.
```

If you can fix this ,will be highly appreciated ,my hunch is t's a problem with the F1 execution mode of udocker ,please help

**Comment by tarunkumard** [comment]
so do you mean udocker doesnt needs java SE environment?

**Comment by eldos-dl** [comment]
What could be the reason behind this? How can this be fixed?

**Comment by mariojmdavid** [comment]
```
hi @tarunkumard 
in my desktop with ubuntu18.04 I manage to run the container, (default execmode P1)
udocker pull denvazh/gatling (OK)

$ udocker run --rm denvazh/gatling 
Warning: non-existing user will be created
 
 ****************************************************************************** 
 *                                                                            * 
 *               STARTING 23957c8b-0ff4-3f82-914c-3a34b3bb8b13                * 
 *                                                                            * 
 ****************************************************************************** 
 executing: gatling.sh
GATLING_HOME is set to /opt/gatling
Choose a simulation number:
     [0] computerdatabase.BasicSimulation
     [1] computerdatabase.advanced.AdvancedSimulationStep01
     [2] computerdatabase.advanced.AdvancedSimulationStep02
     [3] computerdatabase.advanced.AdvancedSimulationStep03
     [4] computerdatabase.advanced.AdvancedSimulationStep04
     [5] computerdatabase.advanced.AdvancedSimulationStep05
0
Select run description (optional)
Simulation computerdatabase.BasicSimulation started...
================================================================================
2020-03-06 12:01:48                                           5s elapsed
---- Requests ------------------------------------------------------------------
> Global                                                   (OK=2      KO=0     )
> request_1                                                (OK=1      KO=0     )
> request_1 Redirect 1                                     (OK=1      KO=0     )

---- Scenario Name -------------------------------------------------------------
[--------------------------------------------------------------------------]  0%
          waiting: 0      / active: 1      / done: 0     
================================================================================

```

**Comment by mariojmdavid** [comment]
futher
I create a container and setup execmode to F1, and obtained the same error, with F3 I got an exception:

 executing: gatling.sh
GATLING_HOME is set to /opt/gatling
Choose a simulation number:
     [0] computerdatabase.BasicSimulation
     [1] computerdatabase.advanced.AdvancedSimulationStep01
     [2] computerdatabase.advanced.AdvancedSimulationStep02
     [3] computerdatabase.advanced.AdvancedSimulationStep03
     [4] computerdatabase.advanced.AdvancedSimulationStep04
     [5] computerdatabase.advanced.AdvancedSimulationStep05
0
Select run description (optional)

12:07:11.374 [ERROR] i.g.h.u.SslContextsFactory - OpenSSL is enabled in the Gatling configuration but it's not available on your architecture.
12:07:11.584 [WARN ] i.n.c.DefaultChannelId - Failed to find the current process ID from ''; using a random value: -784920946
12:07:11.588 [WARN ] i.n.u.c.DefaultPromise - An exception was thrown by io.gatling.http.client.impl.DefaultHttpClient$$Lambda$367/1119712778.operationComplete()
java.lang.NoClassDefFoundError: Could not initialize class io.netty.channel.DefaultChannelId
	at io.netty.channel.AbstractChannel.newId(AbstractChannel.java:100)
	at io.netty.channel.AbstractChannel.<init>(AbstractChannel.java:72)
	at io.netty.bootstrap.FailedChannel.<init>(FailedChannel.java:33)
	at io.netty.bootstrap.AbstractBootstrap.initAndRegister(AbstractBootstrap.java:319)
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect(Bootstrap.java:159)
	at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:152)

(will check with @jorge-lip main devel)

**Comment by tarunkumard** [comment]
sure please do and fix it and let me know in which branch or fork it got fixed

**Comment by eldos-dl** [comment]
@mariojmdavid Is there any quick fix or workaround for F1 or F3 modes
I get the same issue when trying to work with java.
```
udocker create --name=java openjdk:8-jdk-alpine
udocker setup --execmode=F1 java
udocker run java javac -version
```

**Comment by jorge-lip** [comment]
Fixed in 1.1.6

**Comment by DilLip-Chowdary-Codes** [comment]
#### Input

Getting the same issue in the latest version (`1.3.1`) as well.

`udocker create --name=java openjdk:8-jdk-alpine`

`udocker setup --execmode=F1 java`

`udocker run java javac -version`

#### Output:

```sh
Warning: non-existing user will be created
 
 ############################################################################## 
 #                                                                            # 
 #               STARTING 3af1d0be-487a-3186-976d-76af53628543                # 
 #                                                                            # 
 ############################################################################## 
 executing: javac
Error: could not find libjava.so
Error: Could not find Java SE Runtime Environment.
```

#### Version

`udocker --version`

```sh
version: 1.3.1
tarball: https://download.ncg.ingrid.pt/webdav/udocker/udocker-englib-1.2.8.tar.gz https://raw.githubusercontent.com/jorge-lip/udocker-builds/master/tarballs/udocker-englib-1.2.8.tar.gz
tarball_release: 1.2.8
```
